contracts/tests/HotfixClaimRevenue.t.sol

```diff
@@ -0,0 +1,334 @@
+        emit log_named_uint("operator tokens USDC", spigotOperatorTokens);
+        uint256 expectedOwnerTokens = (claimed * revenueSplit) / 100;
+        assertEq(spigotOwnerTokens, expectedOwnerTokens);
+        assertEq(spigotOperatorTokens, claimed - expectedOwnerTokens);
+        vm.stopPrank();
+
+        // 16_685_641: arbiter calls claimAndTrade ( 13500000 in)
+        _rollAndWarpToBlock(16_685_641); 
+        vm.startPrank(arbiter);
+        emit log_string("=> claimAndTrade()");
+        line.claimAndTrade(USDC,hex"d9627aa400000000000000000000000000000000000000000000000000000000000000800000000000000000000000000000000000000000000000000000000000cdfe60000000000000000000000000000000000000000000000000b9c4f4da1409985600000000000000000000000000000000000000000000000000000000000000010000000000000000000000000000000000000000000000000000000000000002000000000000000000000000a0b86991c6218b36c1d19d4a2e9eb0ce3606eb480000000000000000000000006b175474e89094c44da98b954eedeac495271d0f869584cd000000000000000000000000e9039a6968ed998139e023ed8d41c7fa77b7ff7a0000000000000000000000000000000000000000000000cb3b1dde6d63f65891");
+        spigotOwnerTokens = spigot.getOwnerTokens(USDC);
+        spigotOperatorTokens = spigot.getOperatorTokens(USDC);
+        emit log_named_uint("owner tokens USDC after claimAndTrade", spigotOwnerTokens);
+        emit log_named_uint("operator tokens USDC after claimAndTrade", spigotOperatorTokens);
+        vm.stopPrank();
+
+        // 16_685_974: borrower calls useAndRepay
+        _rollAndWarpToBlock(16_685_974); 
+        (, uint256 principal,uint256 interest,,,,,) = line.credits(line.ids(0));
+        emit log_named_uint("principal", principal);
+        emit log_named_uint("interest", interest);
+        
+        vm.startPrank(borrower);
+        emit log_string("=> useAndRepay()");
+        line.useAndRepay(10 ether);
+        vm.stopPrank();
+
+        spigotOwnerTokens = spigot.getOwnerTokens(USDC);
+        spigotOperatorTokens = spigot.getOperatorTokens(USDC);
+        assertEq(IERC20(USDC).balanceOf(address(spigot)), spigotOperatorTokens);
+        emit log_named_uint("owner tokens USDC after useAndRepay", spigotOwnerTokens);
+        emit log_named_uint("operator tokens USDC after useAndRepay", spigotOperatorTokens);
+        emit log_named_uint("line USDC balance", IERC20(USDC).balanceOf(lineAddress));
+        emit log_named_uint("spigot USDC balance", IERC20(USDC).balanceOf(address(spigot)));
+
+        // full 15 USDC has been used at this point
+        /// note: 1.5 USDC remains in operator tokens when bug is present
+
+        // 16_687_230: claimRevenue (called by thomas)
+        _rollAndWarpToBlock(16_687_230); 
+        vm.startPrank(THOMAS);
+        emit log_string("=> claimRevenue()");
+        vm.expectRevert(ISpigot.NoRevenue.selector);
+        claimed = spigot.claimRevenue(BORROWER_REVENUE_EOA, USDC, bytes(""));
+        vm.stopPrank();
+
```

Comment:
this should be where the bug is introduced - on the push payment claimRevenue while there are currently operator tokens. Can u add more tests either way to confirm if spigot is working properly here